### PR TITLE
created a stop command for AtlasMap #49

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

**Comment by lhein** [comment]
good catch

**Comment by lhein** [comment]
fixed the issue with the disposed server output channel

**Comment by lhein** [comment]
I added a test which starts, then stops and then again starts the server. Unfortunately spying on the createOutputChannel methods seems to not work for unknown reasons so I wasn't able to check if that method was called twice.

**Comment by apupier** [comment]
> I added a test which starts, then stops and then again starts the server.

great

> Unfortunately spying on the createOutputChannel methods seems to not work for unknown reasons so I wasn't able to check if that method was called twice.

I take a look if I can find something. I imagine that I can try to spy on the Open the browser.

**Comment by lhein** [comment]
Maybe u can, but it won't test more than we currently already do. Spying on the createOutputChannel would have been used to check if there were 2 calls to it which would basically ensure the channel is created each time. Opening the browser is less of a reliable check.

**Comment by apupier** [comment]
> Maybe u can, but it won't test more than we currently already do. Spying on the createOutputChannel would have been used to check if there were 2 calls to it which would basically ensure the channel is created each time

provided PR using a spy on createOutputChannel https://github.com/lhein/vscode-atlasmap/pull/4

> Opening the browser is less of a reliable check.

I don't get why it would be less reliable? I think that both are interesting check. Checking for "opn" to be called with right values would a good addition (in another PR). Checking that the open is called is checking a more important feature than checking that the output channel is available. i would say not calling open command is "critical", not having the "output channel" is "major".

**Comment by lhein** [comment]
I did not write its less reliable. I just said that spying the open browser command adds not more reliability to the test in comparison to what we already have in place. The PR you provided has some flaws which make the test pass even if it shouldn't. As I said, I tried to spy on that function already but it did not work as expected.

File: src/test/command.test.ts [hunk 1]
```diff
@@ -32,9 +34,14 @@ describe("AtlasMap/Commands", function() {
 		executeCommandSpy.restore();
 		showInformationMessageSpy.restore();
 		showWarningMessageSpy.restore();
+		createOutputChannelSpy.restore();
 		sandbox.restore();
 	});
 
+	beforeEach(function() {
```
Comment (apupier) [inline src/test/command.test.ts]: ah cool a beforeEach
so:
- Mocha before equivalent @beforeClass in Junit
- Mocha beforeEach equivalent to @before in Junit

that's good to know!

Comment (lhein) [reply src/test/command.test.ts]: You can even have those functions in each suite I think.

**Pull request merged**